javagen/src/main/java/com/azure/autorest/template/ServiceSyncClientTemplate.java

```diff
@@ -129,25 +130,21 @@ protected void writeClass(AsyncSyncClient syncClient, JavaClass classBlock, Java
   protected void writeMethods(AsyncSyncClient syncClient, JavaClass classBlock) {
     final ServiceClient serviceClient = syncClient.getServiceClient();
     final MethodGroupClient methodGroupClient = syncClient.getMethodGroupClient();
-    final boolean wrapServiceClient = methodGroupClient == null;
-    if (wrapServiceClient) {
-      serviceClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
-    } else {
-      methodGroupClient.getClientMethods().stream()
-          .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
-          .filter(clientMethod -> !clientMethod.isImplementationOnly())
-          .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
-          .forEach(clientMethod -> {
-            writeMethod(clientMethod, classBlock);
-          });
+
+    List<ClientMethod> clientMethods = serviceClient.getClientMethods();
+    if (CoreUtils.isNullOrEmpty(clientMethods)) {
+      clientMethods = methodGroupClient.getClientMethods();
     }
 
+    clientMethods.stream()
+        .filter(clientMethod -> clientMethod.getMethodVisibility() == JavaVisibility.Public)
+        .filter(clientMethod -> !clientMethod.isImplementationOnly())
+        .filter(clientMethod -> !clientMethod.getType().name().contains("Async"))
+        .filter(clientMethod -> clientMethod.getMethodVisibility().equals(JavaVisibility.Public))
+        .forEach(clientMethod -> {
+          writeMethod(clientMethod, classBlock);
+        });
+
```

Comment:
The CI failure in cadl-java maybe caused by this code change. Previously it checks whether it wraps a ServiceClient or MethodGroupClient, and add client methods accordingly.

And you can see in CI diff only happen to the Sync client, not the Async client.